### PR TITLE
add disable_os_default recipe

### PR DESCRIPTION
**Pull request by alepore**
When using redis from the ubuntu package, i use this simple recipe to disable the default OS init script.
Otherwise i can't use the init script provided by this recipe after a reboot.

Do you think something like this can be a useful addition?

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -16,8 +16,9 @@
 recipe "redisio::configure", "This recipe is used to configure redis by creating the configuration files and init scripts"
 recipe "redisio::sentinel", "This recipe is used to configure redis sentinels by creating the configuration files and init scripts"
 recipe "redisio::enable", "This recipe is used to start the redis instances and enable them in the default run levels"
-recipe "redisio::disable", "this recipe is used to stop the redis instances and disable them in the default run levels"
-recipe "redisio::redis_gem", "this recipe will install the redis ruby gem into the system ruby"
+recipe "redisio::disable", "This recipe is used to stop the redis instances and disable them in the default run levels"
+recipe "redisio::redis_gem", "This recipe will install the redis ruby gem into the system ruby"
+recipe "redisio::disable_os_default", "This recipe can be used to disable the default OS redis init scrip"
```
Comment (shortdudey123) [inline metadata.rb]: typo at the end `scrip`

File: recipes/disable_os_default.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# disable the default OS redis init script
+service_name = case node['platform']
+when 'debian', 'ubuntu'
+  'redis-server'
+end
```
Comment (shortdudey123) [inline recipes/disable_os_default.rb]: `redis` for rhel hosts

**Comment by alepore** [comment]
rebased and updated

**Comment by martinb3** [comment]
:+1: I'm good with this

File: recipes/disable_os_default.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+service_name = case node['platform']
+when 'debian', 'ubuntu'
+  'redis-server'
+when 'redhat', 'centos', 'fedora', 'scientific', 'suse', 'amazon'
+  'redis'
+end
+
+if service_name
+  service service_name do
+    action :disable
```
Comment (shortdudey123) [inline recipes/disable_os_default.rb]: `action [ :stop, :disable ]`

File: recipes/disable_os_default.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+# disable the default OS redis init script
+service_name = case node['platform']
+when 'debian', 'ubuntu'
+  'redis-server'
+when 'redhat', 'centos', 'fedora', 'scientific', 'suse', 'amazon'
+  'redis'
+end
+
+if service_name
```
Comment (shortdudey123) [inline recipes/disable_os_default.rb]: Switch to `only_if` not an `if` block

**Comment by devsibwarra** [comment]
Hit this earlier today.  Added the service resource with suggestions by @shortdudey123 to my wrapper script.

This PR probably needs a re-base, then 👍  from me

**Comment by brianbianco** [comment]
@alepore Can you rebase this for a merge ?

**Comment by alepore** [comment]
updated and rebased!

**Comment by shortdudey123** [comment]
```
Inspecting 1 file
W

Offenses:

recipes/disable_os_default.rb:23:1: C: Style/CaseIndentation: Indent when as deep as case.
when 'debian', 'ubuntu'
^^^^
recipes/disable_os_default.rb:25:1: C: Style/CaseIndentation: Indent when as deep as case.
when 'redhat', 'centos', 'fedora', 'scientific', 'suse', 'amazon'
^^^^
recipes/disable_os_default.rb:27:1: W: Lint/EndAlignment: end at 27, 0 is not aligned with case at 22, 15.
end
^^^

1 file inspected, 3 offenses detected
```

**Comment by alepore** [comment]
rubocop 🆗

**Comment by shortdudey123** [comment]
awesome! looking good to me :)

**Comment by brianbianco** [comment]
Thanks all involved for pushing this through.

**Pull request merged**